src/general/WorldGenerationSettings.cs

```diff
@@ -1,6 +1,158 @@
-﻿/// <summary>
+﻿using System;
+
+/// <summary>
 ///   Player configurable options for creating the game world
 /// </summary>
 public class WorldGenerationSettings
 {
+    public enum LifeOrigin
+    {
+        Vent,
+        Pond,
+        Panspermia,
+    }
+
+    public enum PatchMapType
+    {
+        Procedural,
+        Classic,
+    }
+
+    /*
+    Values for this particular object
+    */
+
+    public bool Lawk { get; set; }
```

Comment:
Most of these are pretty self explanatory, but could you add a short summary for each? These are pretty widely affecting things so similarly to the game settings object, I think all of these properties should be documented.

Also I think this field should be named LAWK.

Reply:
Code styling had a problem with the name LAWK so I changed it to this.

Reply:
You need to add the LAWK as an abbreviation so that it knows that it is a single word and stops complaining.

Reply:
Thrive.sln.DotSettings around line 460